x-pack/plugins/security_solution/server/lib/detection_engine/signals/threshold/build_threshold_aggregation.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ThresholdNormalized } from '../../../../../common/detection_engine/schemas/common';
+import { shouldFilterByCardinality } from './utils';
+
+export const buildThresholdMultiBucketAggregation = ({
+  threshold,
+  timestampField,
+  sortKeys,
+}: {
+  threshold: ThresholdNormalized;
+  timestampField: string;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  sortKeys: Record<string, any> | undefined;
+}) => {
+  return {
+    thresholdTerms: {
+      composite: {
+        sources: threshold.field.map((term, i) => ({
+          [term]: {
+            terms: {
+              field: term,
+            },
+          },
+        })),
+        after: sortKeys,
```

Comment:
consider coalescing to `null` here, i.e.

```ts
after: sortKeys ?? null,
```

, because per the unit tests, when `sortKeys` is `undefined`:

```ts
    it('Generates aggregation with cardinality', async () => {
      const thresholdWithCardinality = {
        ...threshold,
        cardinality: [{ field: 'destination.ip', value: 2 }],
      };
      const aggs = buildThresholdMultiBucketAggregation({
        threshold: thresholdWithCardinality,
        timestampField,
        sortKeys: `undefined`, // <-- this will result in `after` being `undefined` in the JSON
      });
      // ...
```

the `after` field is `undefined` in the resulting JSON, (which is invalid JSON):

```ts
          composite: {
            after: undefined, // <-- invalid JSON
            size: 10000,
            sources: [
              {
                'host.name': {
                  terms: {
                    field: 'host.name',
                  },
                },
              },
              {
                'source.ip': {
                  terms: {
                    field: 'source.ip',
                  },
                },
              },
            ],
          },
```